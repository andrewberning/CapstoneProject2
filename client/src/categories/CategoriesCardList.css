.CategoriesCardList {
  margin: 0 auto;
  max-width: 1000px;
}

.CategoriesCardList .categories-list {
  text-decoration: none;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  padding: 0;
  margin: 0;
}

.CategoriesCardList .categories-list li {
  flex: 1 1 calc(33.33% - 20px); 
  max-width: calc(33.33% - 20px);
  margin: 10px;
}

/* 2 cards per row on smaller screens */
@media (max-width: 1080px) {
  .CategoriesCardList .categories-list li {
    flex: 1 1 calc(50% - 20px); 
    max-width: 300px;
  }
}

/* 2 cards per row on smaller screens */
@media (max-width: 768px) {
  .CategoriesCardList .categories-list li {
    flex: 1 1 calc(50% - 20px); 
    max-width: calc(50% - 20px);
  }
}

/* 1 card per row on the smallest screens */
@media (max-width: 480px) {
  .CategoriesCardList .categories-list li {
    flex: 1 1 100%; 
    max-width: 100%;
  }
}
