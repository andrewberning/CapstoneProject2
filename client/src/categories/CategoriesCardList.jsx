import CategoriesCard from "./CategoriesCard"

export default function CategoriesCardList({ categories }) {
  return (
    <div className="ClothingCardList d-flex justify-content-center gap-4">
      {categories.map(category => (
        <CategoriesCard
          key={category.id}
          category={category.name}
        />
      ))}
    </div>
  );
}