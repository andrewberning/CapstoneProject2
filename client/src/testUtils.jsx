import React from "react";
import UserContext from "./auth/UserContext";
import { vi } from 'vitest';

const demoUser = {
  username: "testuser",
  first_name: "Test",
  last_name: "User",
  email: "testuser123@test.net",
};

// Mock functions
const mockSetCurrentUser = vi.fn();
const mockLogin = vi.fn();
const mockLogout = vi.fn();
const mockSignup = vi.fn();
const mockAddToCart = vi.fn();
const mockHasAddedToCart = vi.fn();

const UserProvider = ({ 
  children,
  currentUser = demoUser,
  setCurrentUser = mockSetCurrentUser,
  login = mockLogin,
  logout = mockLogout,
  signup = mockSignup,
  addToCart = mockAddToCart,
  hasAddedToCart = mockHasAddedToCart,
  cartItems = [],
  setCartItems = () => {},
  categories = [],
  infoLoaded = true,
  totalAmount = 0,
  totalItems = 0,
}) => (
  <UserContext.Provider value={{ 
    currentUser,
    setCurrentUser,
    login,
    logout,
    signup,
    addToCart,
    hasAddedToCart,
    cartItems,
    setCartItems,
    categories,
    infoLoaded,
    totalAmount,
    totalItems,
  }}>
    {children}
  </UserContext.Provider>
);

export { UserProvider, mockSetCurrentUser, mockLogin, mockLogout, mockSignup, mockAddToCart, mockHasAddedToCart };
