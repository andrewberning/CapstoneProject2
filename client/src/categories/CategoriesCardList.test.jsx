import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import CategoriesCardList from "../banner/CategoriesBannerList";
import { UserProvider } from "../testUtils";

describe('CategoriesCardList component', () => {
  it('renders without crashing', () => {
    render(
      <MemoryRouter>
        <UserProvider >
          <CategoriesCardList />
        </UserProvider>
      </MemoryRouter>
    );
  });

  it("matches snapshot", function () {
    const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <CategoriesCardList />
        </UserProvider>
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});