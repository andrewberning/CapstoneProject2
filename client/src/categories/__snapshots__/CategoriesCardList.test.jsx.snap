// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CategoriesCardList component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="CategoriesBannerList"
  >
    <ul
      class="categories-list"
    />
  </div>
</DocumentFragment>
`;
